### ISSUE TITLE
[Documentation Update]: Module names - clarity

### ISSUE THREAD

**Issue by jbinko** [Azure/Azure-Verified-Modules#1621]
### Check for previous/existing GitHub issues

- [x] I have checked for previous/existing GitHub issues

### Description

The following sections describe the naming conventions for modules, but contributors are not required to create these names themselves. Instead, we should direct them to the csv link where they can find the pre-assigned module names.
https://azure.github.io/Azure-Verified-Modules/specs/shared/#naming-convention
https://azure.github.io/Azure-Verified-Modules/specs/shared/#bicep-resource-module-naming
https://azure.github.io/Azure-Verified-Modules/specs/shared/#bicep-pattern-module-naming

I suggest adding notes in the links above with pointer to csv to recommend that contributors utilize existing module names.

You can find the existing names of modules here: https://github.com/Azure/Azure-Verified-Modules/blob/main/docs/static/module-indexes/BicepResourceModules.csv

This approach prevents any confusion for new contributors attempting to devise module names based on complex naming conventions.

**Comment by eriqua** [comment]
Hi @jbinko, thanks for raising this issue.

Module names are not preassigned, but provided by contributors raising a module proposal. 
Only after that the module name will be listed in the csv file. 
Hence, contributors should be aware of the naming convention before the name is listed.

Could you please elaborate more on which part of the documentation is unclear in that sense?
Exactly which phase of the contribution are we talking about (e.g., before the module proposal/after the proposal but before the first release/update of an existing module)?

Thanks!

**Comment by jbinko** [comment]
From my experience, while the owner may suggest a module name, it can be changed. The key information for beginning implementation is found in the CSV file. This file contains essential data assigned to you after the initial proposal. The idea is to simplify the process for contributors, allowing them to use the CSV file rather than having to fully comprehend all the details in the specification documents. This approach can be highly beneficial for contributors.

https://github.com/Azure/Azure-Verified-Modules/blob/main/docs/static/module-indexes/BicepResourceModules.csv

IMO, we should emphasize and incorporate these points in the documentation - including the link.

**Comment by eriqua** [comment]
Thanks a lot @jbinko for elaborating. 
Would you see it useful then to add link and details about the csv to the contribution guide rather than the specs? Would you see this issue related to #1600?

**Comment by jbinko** [comment]
Yeah, including a link in the spec and more details about it could be beneficial. We aim to encourage the community to read the specs thoroughly. Adding more clarity and hints within the spec could improve understanding, in my opinion.

**Comment by eriqua** [comment]
Cc @matebarabas @ReneHezser

**Comment by matebarabas** [comment]
Hi @jbinko, thanks for calling this out - all good suggestions. You're right, the module naming process (while following the linked specifications) is often an iterative process. While it's pretty straightforward for (most) resource modules, it may require some additional considerations in case of pattern and utility modules. Ultimately, the module's approved name is captured both in the module proposal issue AND the related module index page (backed by the corresponding CSV), but your suggestion on making this clear(er) in the specs makes sense. I'll follow up on this.
